### PR TITLE
Added support for async await in handler

### LINKED ISSUE #52: Support for async/await in handler
Since version `0.3.0` we have support for `async/await` in middlewares.

We could potentially apply the same approach to the handler. Since the handler is invoked and managed by the middy wrapper, we could support different syntaxes including handlers returning promises and consequently `async/await` handlers.

These are the two syntaxes I envision might be cool to support:

```javascript
const handler = middy((event, context) => {
  return Promise.resolve({
    statusCode: 200,
    body: 'hello'
  })
})
```

and, with `async/await`:

```javascript
const handler = middy(async (event, context) => {
  return {
    statusCode: 200,
    body: 'hello'
  }
})
```

Any thoughts?

I think this is feasible by addressing the handler runtime logic [here](https://github.com/middyjs/middy/blob/master/src/middy.js#L154-L168).

### PR DESCRIPTION
**Pull request by lmammino**
Closes #52 

TODO:

- [x] Add docs and examples
- [x] Update typings for typescript
- [x] Version bump

**READY FOR REVIEW** 🤓

### CODE REVIEW

**Comment by lmammino** [comment]
@joseSantacruz can you please verify if the typescript changes here make sense?

File: README.md [hunk 1]
```diff
@@ -249,6 +249,45 @@ to the user.
...
+    .then(() => {
+      return someOtherAsyncStuff()
+    })
+    .then(() => {
+      callback(null, {foo: bar})
+    })
+})
+```
+
+We believe that the async/await version is easier to read and reason about.
```
Comment (peterjcaulfield) [inline README.md]: This sentence is redundant based on sentence above I think.

File: src/__tests__/middy.js [hunk 2]
```diff
@@ -405,6 +405,63 @@ describe('🛵  Middy test suite', () => {
...
+      throw new Error('bad stuff happened')
+    })
+
+    handler({}, {}, (err, response) => {
+      expect(err.message).toEqual('bad stuff happened')
+      endTest()
+    })
+  })
+
+  test('An handler that returns a non-promise should trigger an error', (endTest) => {
```
Comment (peterjcaulfield) [inline src/__tests__/middy.js]: typo. s/An handler/A handler

File: README.md (hunk 1 above)
Comment (lmammino) [reply README.md]: Good point!

File: src/__tests__/middy.js (hunk 2 above)
Comment (lmammino) [reply src/__tests__/middy.js]: Thanks

Comment (dkatavic) [inline src/__tests__/middy.js]: What about handling then-able objects?

**Pull request merged**